src/ast/ast_validations.c

```diff
@@ -1193,7 +1194,14 @@ static VISITOR_STRATEGY _Validate_set_property
 
 	const cypher_astnode_t *ast_prop = cypher_ast_set_property_get_property(n);
 	const cypher_astnode_t *ast_entity = cypher_ast_property_operator_get_expression(ast_prop);
-	if(cypher_astnode_type(ast_entity) != CYPHER_AST_IDENTIFIER) {
+	cypher_astnode_type_t type = cypher_astnode_type(ast_entity);
+	if(type == CYPHER_AST_PROPERTY_OPERATOR) {
+		AR_ExpNode* node = AR_EXP_FromASTNode(ast_entity);
```

Comment:
in vctx->define_aliases we add to the value of the rax if this alias is node or edge so no need to build expression and evaluate it

Reply:
I missed that. Please check if the last change is correct. Thank you.